src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -380,3 +400,60 @@ export function isPublicMethodSymbol(
 export function generateDotNetLambdaHandler(components: DotNetLambdaHandlerComponents): string {
     return `${components.assembly}::${components.namespace}.${components.class}::${components.method}`
 }
+
+export async function installDebugger({
+    runtime,
+    codeUri
+}: {
+    runtime: string,
+    codeUri: string
+}): Promise<string> {
+    const vsdbgPath = path.resolve(codeUri, '.vsdbg')
+
+    try {
+        await access(vsdbgPath)
+
+         // vsdbg is already installed.
+        return vsdbgPath
+    } catch {
+        // We could not access vsdbgPath. Swallow error and continue.
+    }
+
+    try {
+        await mkdir(vsdbgPath)
+
+        const process = crossSpawn(
+            'docker',
+            [
+                'run',
+                '--rm',
+                '--mount',
+                `type=bind,src=${vsdbgPath},dst=/vsdbg`,
+                '--entrypoint',
+                'bash',
+                `lambci/lambda:${runtime}`,
+                '-c',
+                '"curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg"'
+            ],
+            {
+                windowsVerbatimArguments: true
+            }
+        )
+
+        await new Promise<void>((resolve, reject) => {
```

Comment:
I intentionally avoided using `utilities/childProcess`, because it is too heavyweight for this purpose--we don't need the overhead of capturing `stderr`/`stdout`.